File: engineering/codebase/smartqr.frontend-services/src/domain/codes/content/models/SmsContent.ts
import { ContentType } from "../enums/ContentType";

/** Represents sms content — opens the scanner's SMS composer addressed to the number. */
export interface SmsContent {
  /** The content-type discriminator. */
  type: typeof ContentType.Sms;

  /** The recipient phone number. */
  phone: string;

  /** The message text prefilled in the composer. */
  message?: string;
}

File: engineering/codebase/smartqr.backend-services/SmartQr.Domain/Codes/Content/Sms/Models/SmsContent.cs
namespace SmartQr.Domain.Codes.Content.Sms.Models;

/// <summary>Pre-filled SMS — encodes to <c>SMSTO:phone</c>, or <c>SMSTO:phone:message</c> when a message is supplied.</summary>
public sealed record SmsContent : CodeContent
{
    /// <summary>Recipient phone number.</summary>
    public required string Phone { get; init; }

    /// <summary>Message text prefilled in the composer.</summary>
    public string? Message { get; init; }

    /// <inheritdoc />
    public override string Encode()
    {
        var phone = ContentEncoding.Clean(Phone);
        var message = ContentEncoding.Clean(Message);
        return message.Length > 0 ? $"SMSTO:{phone}:{message}" : $"SMSTO:{phone}";
    }
}
